src/index.js

```diff
@@ -389,6 +402,8 @@ class SoshContentScript extends ContentScript {
     await this.navigateToPersonalInfos()
     await this.runInWorker('getIdentity')
     await this.saveIdentity(this.store.infosIdentity)
+    // Logging out every run to avoid in between scenarios and sosh/orange mismatched sessions
+    await this.logout()
```

Comment:
If any error is thrown, this.logout won't be called. If you want to be sure that the logout function is called, you should call it in the beginning of the execution.

Reply:
Yep, it is handle in the navigateToLoginForm method, at the start of the execution to cover this case. That way we just have to handle 3 scenarios, thefirst execution, the autoLogin, and the logout + autoLogin for the most part to avoid orange/sosh mixing sessions 👍